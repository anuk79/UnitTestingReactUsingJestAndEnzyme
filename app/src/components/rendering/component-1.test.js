
// vendors
import React from "react";
import { shallow, mount } from 'enzyme';
import { shallowToJson } from "enzyme-to-json";

// components
import ComponentOne from './component-1';

describe("Component One", () => {
    it("should render correctly", () => {
        const wrapper = shallow(<ComponentOne />);
        expect(shallowToJson(wrapper)).toMatchSnapshot();
        expect(wrapper.find('h1').exists()).toBeTruthy();
        expect(wrapper.find('h1').length).toBe(1);
        expect(wrapper.find('label').exists()).toBeTruthy();
        expect(wrapper.find('.username').exists()).toBeTruthy();
        expect(wrapper.find('ComponentTwo').exists()).toBeTruthy();
        expect(wrapper.find('ComponentTwo').prop('errorMessage')).toEqual('invalid user');
        expect(wrapper.find('ComponentTwo').props().errorMessage).toEqual('invalid user');
        expect(wrapper.find('ComponentTwo').props().errorMessage).toBe('invalid user');
    });
    
    it("should render correctly with children", () => {
        const wrapper = mount(<ComponentOne />);
        expect(shallowToJson(wrapper)).toMatchSnapshot();
        expect(wrapper.find('h1').exists()).toBeTruthy();
        expect(wrapper.find('h1').length).toBe(1);
        expect(wrapper.find('label').exists()).toBeTruthy();
        expect(wrapper.find('.username').exists()).toBeTruthy();
        expect(wrapper.find('ComponentTwo').exists()).toBeTruthy();
    });

    it('should call handleChange', () => {
        const wrapper = shallow(<ComponentOne />);
        const originalAlert = window.alert;
        const alertMock = jest.fn();
        window.alert = alertMock;

        wrapper.find('ComponentTwo').prop('handleChange')({ target: { value: 'testValue' } });
        expect(alertMock).toBeCalledTimes(1);
        expect(alertMock).toBeCalledWith('The value changed to testValue');

        window.alert = originalAlert;
    });
});