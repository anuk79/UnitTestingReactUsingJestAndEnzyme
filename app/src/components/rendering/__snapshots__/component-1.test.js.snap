// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component One should render correctly 1`] = `
<Fragment>
  <h1>
    Test heading
  </h1>
  <div
    className="username"
  >
    <label>
      Username
    </label>
    <ComponentTwo
      errorMessage="invalid user"
      handleChange={[Function]}
      isError={false}
      username="testUser"
    />
  </div>
</Fragment>
`;

exports[`Component One should render correctly with children 1`] = `
<ComponentOne>
  <h1>
    Test heading
  </h1>
  <div
    className="username"
  >
    <label>
      Username
    </label>
    <ComponentTwo
      errorMessage="invalid user"
      handleChange={[Function]}
      isError={false}
      username="testUser"
    >
      <input
        onChange={[Function]}
        type="text"
        value="testUser"
      />
    </ComponentTwo>
  </div>
</ComponentOne>
`;
